ext/zkrb.c

```diff
@@ -813,22 +824,22 @@ static VALUE method_zkrb_iterate_event_loop(VALUE self) {
   // add our self-pipe to the read set, allow us to wake up in case our attention is needed
   int pipe_r_fd = get_self_pipe_read_fd(self);
 
-  FD_SET(pipe_r_fd, &rfds);
+  rb_fd_set(pipe_r_fd, &rfds);
 
   maxfd = (pipe_r_fd > fd) ? pipe_r_fd : fd;
 
-  rc = rb_thread_select(maxfd+1, &rfds, &wfds, &efds, &tv);
+  rc = rb_thread_fd_select(maxfd+1, &rfds, &wfds, &efds, &tv);
 
   if (rc > 0) {
```

Comment:
I'm not sure why, but in ree this is hitting the 'else' case below and spewing

```
[ERROR] (zkrb.c:857: errno: Bad file descriptor) select returned an error: rc=-1 interest=3 fd=9 pipe_r_fd=5 maxfd=9 irc=0 timeout=6.667000
```

when running the tests. It works fine in 2.2.0.